src/ed25519.rs

```diff
@@ -183,6 +183,12 @@ impl Drop for SecretKey {
     }
 }
 
+impl AsRef<[u8]> for SecretKey {
```

Comment:
I'm not sure if I'm overly paranoid, but elsewhere I've avoided `impl`ing `AsRef` directly on types which wrap secrets like this, preferring explicit APIs for accessing secrets.

That said, this is the sort of thing that could probably benefit from some shared, reusable, and explicit traits for secret-keeping types, e.g. "like `AsRef<[u8]>`, but instead `ExposeSecret<[u8]>`".